plugins/offline/src/main/java/com/mapbox/mapboxsdk/plugins/offline/OfflineDownload.java

```diff
@@ -0,0 +1,131 @@
+package com.mapbox.mapboxsdk.plugins.offline;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import com.mapbox.mapboxsdk.offline.OfflineTilePyramidRegionDefinition;
+
+import java.util.Arrays;
+
+/**
+ * OfflineDownload model wraps the offline region definition with notifications options and the offline region metadata,
+ * is a companion object to OfflineRegion with regionId and DownloadService with serviceId.
+ */
+public class OfflineDownload implements Parcelable {
+
+  public static final String KEY_BUNDLE = "com.mapbox.mapboxsdk.plugins.offline.download.object";
+
+  private final OfflineTilePyramidRegionDefinition definition;
+  private final NotificationOptions notificationOptions;
+  private final byte[] metadata;
+
+  // unique identifier used by service + notifications
+  // will be created at service startup as part of onStartCommand
+  private int serviceId = -1;
+  // unique identifier matching an offline region
+  // will be created during the start of a download after creating an offline region
+  private long regionId = -1;
+  // download progress
+  private int progress;
+
+  OfflineDownload(OfflineTilePyramidRegionDefinition offlineTilePyramidRegionDefinition, byte[] metadata,
+                          NotificationOptions notificationOptions) {
+    this.definition = offlineTilePyramidRegionDefinition;
+    this.metadata = metadata;
+    this.notificationOptions = notificationOptions;
+  }
+
+  private OfflineDownload(Parcel in) {
+    definition = in.readParcelable(OfflineTilePyramidRegionDefinition.class.getClassLoader());
+    serviceId = in.readInt();
+    regionId = in.readLong();
+    notificationOptions = in.readParcelable(NotificationOptions.class.getClassLoader());
+    progress = in.readInt();
+    metadata = new byte[in.readInt()];
+    in.readByteArray(metadata);
+  }
+
+  OfflineTilePyramidRegionDefinition getRegionDefinition() {
+    return definition;
+  }
+
+  byte[] getMetadata() {
```

Comment:
What if we exposed a higher level API for the metadata (can be ticketed separately). Instead of exposing the `byte[]` separately, we'd expose a POJO with, say, title and description, that we serialize/deserialize as part of this plugin?

Reply:
Just saw `OfflineUtils`, it simplifies things but I'd even go further than that and never expose that directly.

Reply:
Yeah, initially I went with String name and afterwards went back for byte[]. I'm liking the idea of  a POJO. To make that work, we will need an object that is Parceable as we need to be able to push that into a Bundle for Intent communication.

Reply:
@tobrun Or a small POJO easily convertible to JSON with Gson and use the String representation? Or would that potentially be too large for a Bundle?